rosserial_client/src/ros_lib/ros/node_handle.h

```diff
@@ -209,6 +230,20 @@ class NodeHandle_ : public NodeHandleBase_
     /* while available buffer, read data */
     while (true)
     {
+      // If a timeout has been specified, check how long spinOnce has been running.
+      if (spin_timeout_ > 0)
+      {
+        // If the maximum processing timeout has been exceeded, exit with error.
+        // The next spinOnce can continue off where it left off, or optionally
+        // based on the application in use, the hardware buffer could be flushed
+        // and start fresh.
+        if ((hardware_.time() - c_time) > spin_timeout_)
+        {
+          // Reset back to first mode.
+          mode_ = MODE_FIRST_FF;
```

Comment:
@mikepurvis, if we intend to allow for just a continuation of processing, setting this would be detrimental. This kind of forces a buffer flush.

Reply:
Might be better to avoid doing so then. All the deserializing state is maintained between spins— if there's outside logic that drops portions of the buffer, the spin will realise this and resynchronize anyway.